### ISSUE TITLE
Wrong fee calculation

### ISSUE THREAD

**Issue by glonlas** [gcarq/freqtrade#176]
Hi guys,

I think the way we calculate fees are wrong in `persistence::calc_profit()`.
Bittrex takes 0.25% (0.0025) on each transaction (buy and sell). But the behavior is different if it is a buy or a sell.

**Buy behavior:** Bittrex will take the fee from your BTC wallet, not deduct it from the trade. That means it always cost us more BTC than the order we made (Add fees on top of the order)
**Buy formula** = `(#unit * open_rate) + ((#unit * open_rate)* 0.0025)`

**Sell behavior:** Bittrex will deduct fees from your order result. That means we always get less than the amount we sell (Deduct fee from the profit/loss)
**Sell formula =** `(#unit * close_rate) - ((#unit * close_rate)* 0.0025)`

**Percentage profit/loss:** `(Sell result / Buy result) - 1`

### Example:
I buy one coin at `0.00258580` (Open rate) then I will sell it at `0.00279266` (Close rate)

Buy = `(1 * 0.00258580) + ((1 * 0.00258580)* 0.0025)` = 0.002592265 BTC
Sell = `(1 * 0.00279266) - ((1 * 0.00279266)* 0.0025)` = 0.002785682 BTC
Profit = `0.002785682 - 0.002592265` = +0.000193417 BTC
Percentage profit/loss = `(0.002785682 / 0.002592265) - 1` = 0.0746 (7.46%)

## Issue in the way fee are calculated in the code
During a Trade by doing `fee=exchange.get_fee() * 2,` we do not apply the right amount a fees.
Source: https://github.com/gcarq/freqtrade/blob/b77fad6e5f204cb3593c33bc650c0317a97b213b/freqtrade/main.py#L233

The formula on `calc_profit()`does not seem right.
Source: https://github.com/gcarq/freqtrade/blob/b77fad6e5f204cb3593c33bc650c0317a97b213b/freqtrade/persistence.py#L110-L119

**Can we have a unit-test to see I am wrong on the fee calculation?**

**Comment by csplkiller** [comment]
https://bittrex.com/Fees
All trades have a 0.25% commission. !!
not 25%, but 0.25% from each order!
But even this small commission is not considered a profit?

**Comment by glonlas** [comment]
My apologies, it was a typo. This is not 25% but 0.25% as you said. However it is not equal to 0.50%.

Let me explain: (I am using dollar for this example, but this is exactly the same with BTC)
You buy $100 and sell at $2,000
With 0.50% (0.25% * 20) of $2000, your fees will be $10

However, in the reality Bittrex will charge you 0.25% on the initial $100 and 0.25% on the $2000.
For $100 = $0.25 (100*0.0025)
For $2000 = $5 (2000*0.0025)

So **your real fees are $5.25, instead of $10**.

On a freqtrade, because we are doing thousand of transaction, this difference can totally chance your ROI

**Comment by pmaster555** [comment]
Yes, counting 0.50% is not right. It is necessary to subtract 0.25 at purchase and 0.25 at sale

**Comment by glonlas** [comment]
do you mind if I update this behavior?

**Comment by gcarq** [comment]
@glonlas Thanks a lot for the comprehensive bug reports, I really appreciate it.
Feel free to create a PR to fix it.
Keep in mind that once a Trade object has been created and the linked LIMIT_BUY order has been fulfilled, the `amount` and `open_rate` are updated accordingly. The same is true for LIMIT_SELL orders. The method responsible for this is here: https://github.com/gcarq/freqtrade/blob/develop/freqtrade/persistence.py#L84

**Comment by glonlas** [comment]
I am working on the PR. You can assign me on it.
